File: Codeforces/CF 906/A_Doremy_s_Paint_3.py
t = int(input())
for _ in range(t):
    n = int(input())
    ar = list(map(int,input().split()))
    my_set = set(ar)
    if(len(my_set)>2):print("No")
    else:
        cnt = ar.count(ar[0])
        rem = n-cnt
        if(abs(cnt-rem)<=1 or cnt==n):print("Yes")
        else: print("No")

File: Codeforces/CF 906/A_Doremy_s_Paint_3.cpp
/*----------------------------------------
    author:  Imam
    created: 24.05.2025 22:08:28
----------------------------------------*/
#include<bits/stdc++.h>
using namespace std;
inline void solve()
{
    int n; cin>>n;
    vector<int> ar(n);
    for(auto &i:ar) cin>>i;
    set<int> s(ar.begin(),ar.end());
    if(s.size()>2)cout<<"No\n";
    else{
        int cnt=count(ar.begin(),ar.end(),ar[0]);
        int rem=n-cnt;
        if(abs(rem-cnt)<=1 || cnt==n)cout<<"Yes\n";
        else cout<<"No\n";
    }
}
int main()
{
    std::ios_base::sync_with_stdio(false);
    std::cin.tie(nullptr); cout.tie(NULL);
    
    int t; cin>>t;
    while(t--) solve();
    return 0;
}